### ISSUE TITLE
Scrolling broken when clip rect contains a non zero x origin

### ISSUE THREAD

**Issue by nc4rrillo** [servo/webrender#1931]
Hi, 
I'm working on a UI framework using WebRender and I've run into a peculiar issue. I'm not sure if its just me misunderstanding the API or if its a genuine bug.

I'm using the scroll API to scroll the first scrollable node found at a given WorldPoint, which so far has been working fine. However, I've been trying to offset the viewport rect to take into account the width of the border, and it has the effect of breaking scrolling.

Good YAML
```
---
root: 
  items: 
    - 
      "clip-rect": [0, 5, 495, 490]
      "clip-and-scroll": 0
      "backface-visible": true
      type: "scroll-frame"
      id: 1
      "content-size": [1495, 1500]
      bounds: [0, 5, 495, 490]
    - 
      bounds: [0, 0, 500, 500]
      "clip-rect": [0, 0, 500, 500]
      "clip-and-scroll": 0
      "backface-visible": true
      type: border
      width: 5
      "border-type": normal
      color: black
      style: solid
    - 
      bounds: [0, 0, 200, 200]
      "clip-rect": [0, 0, 200, 200]
      "clip-and-scroll": 1
      "backface-visible": true
      type: rect
      color: 76.5 204 255 1.0000
  id: [0, 0]
pipelines: []`
```
vs

Bad YAML

````
---
root: 
  items: 
    - 
      "clip-rect": [5, 5, 490, 490]
      "clip-and-scroll": 0
      "backface-visible": true
      type: "scroll-frame"
      id: 1
      "content-size": [1495, 1500]
      bounds: [5, 5, 490, 490]
    - 
      bounds: [0, 0, 500, 500]
      "clip-rect": [0, 0, 500, 500]
      "clip-and-scroll": 0
      "backface-visible": true
      type: border
      width: 5
      "border-type": normal
      color: black
      style: solid
    - 
      bounds: [0, 0, 200, 200]
      "clip-rect": [0, 0, 200, 200]
      "clip-and-scroll": 1
      "backface-visible": true
      type: rect
      color: 76.5 204 255 1.0000
  id: [0, 0]
pipelines: []
``````

Notice how the clip-rect for the scroll-frame is offset by 5 in the second YAML dump. Wrench doesn't support scrolling input while viewing dumps, but the API I use is here:

```
                self.render_api.scroll(
                    self.document_id,
                    webrender_api::ScrollLocation::Delta(webrender_api::LayoutVector2D::new(0., delta_y)),
                    self.mouse_position,
                    phase
                );
```

Basically I'm trying to accomplish having the viewport of the scrollable frame take the border of the outer rect into account, so that when the contents of the scroll frame are scrolled they don't render over the border.

**Comment by glennw** [comment]
cc @mrobinson

**Comment by nc4rrillo** [comment]
After digging through the scrolling code in WR (fascinating stuff btw), I found this issue and it ended up being in....my code.

Closing.
